.github/workflows/ci.yml

```diff
@@ -140,3 +140,31 @@ jobs:
       with:
         token: ${{ secrets.GITHUB_TOKEN }}
         args: --workspace --all-features --all-targets -- -D warnings
+  minimal_versions:
+    name: Minimal versions
+    needs: smoke
+    strategy:
+      matrix:
+        os: [ "ubuntu-latest", "windows-latest", "macos-latest" ]
+        rust: [ "1.54.0" ] # MSRV
+    continue-on-error: ${{ matrix.rust != 'stable' }}
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+         profile: minimal
+         toolchain: nightly
+      - uses: actions-rs/toolchain@v1
+        with:
+         profile: minimal
+         toolchain: ${{ matrix.rust }}
+         override: true
+      - name: Downgrade crates to minimal supported versions
+        run: cargo +nightly update -Z minimal-versions
```

Comment:
I've heard a good number of horror stories from dependencies not testing this, breaking people who do want this.

I'm uncomfortable committing to this unless all dependencies being used by this job either are also testing this or explicitly document as a contract that they are zero-dependency crates.

Reply:
@epage but why? You will hit this either way from downstream crates, sooner or later, like this time has happened. It's only a matter of catching regressions earlier.

Reply:
Keep in mind I'm going off of vague memory of comments I've seen on reddit.  I remember reading of some people (like burntsushi?) trying to use this but core Rust crate authors were refusing to verify it, making it a minefield for dependents to verify and it not break them.

In breaking this down, nothing breaks until I upgrade my minimal versions, unlike MSRV.  The downside is when I do upgrade, it becomes a chore of chasing through my dependency chain to find all of the issues.  This becomes an impediment to doing what should be a basic operation.

Without my dependencies also being on board, I'm not seeing how the benefits outweigh the costs.

Reply:
> I'm uncomfortable committing to this unless all dependencies being used by this job either are also testing this or explicitly document as a contract that they are zero-dependency crates.

In case some of your transitive dependencies have too low minimal version, you can just specify them in `Cargo.toml`, while not using in the crate by yourself, so I don't see where the `horror` stories can come from.

Reply:
Having to micro-manage this might not sound too big of a deal to you but it can be a deal to others, especially when they are managing a significant number of crates.

As I said elsewhere, we can move forward with the dependency updates as we work through how to handle the workflows.  We can decouple the fix from the effort of either trying to get buy-in from dependencies or finding a compromise for how to handle CI.